js/mage/adminhtml/wysiwyg/tinymce/setup.js

```diff
@@ -0,0 +1,361 @@
+/**
+ * OpenMage
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE_AFL.txt.
+ * It is also available at https://opensource.org/license/afl-3-0-php
+ *
+ * @category    Mage
+ * @package     Mage_Adminhtml
+ * @copyright   Copyright (c) 2006-2020 Magento, Inc. (https://www.magento.com)
+ * @copyright   Copyright (c) 2018 The OpenMage Contributors (https://www.openmage.org)
+ * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+var tinyMceWysiwygSetup = Class.create();
+tinyMceWysiwygSetup.prototype =
+{
+    mediaBrowserCallback: null,
+    mediaBrowserMetal: null,
+    mediaBrowserValue: null,
+
+    openmagePluginsOptions: $H({}),
+
+    initialize: function (htmlId, config) {
+        this.id = htmlId;
+        this.selector = 'textarea#' + htmlId;
+        this.config = config;
+        varienGlobalEvents.attachEventHandler('tinymceChange', this.onChangeContent.bind(this));
+
+        if (typeof tinyMceEditors === 'undefined') {
+            window.tinyMceEditors = $H({});
+        }
+        tinyMceEditors.set(this.id, this);
+    },
+
+    setup: function (mode) {
+        var self = this;
+
+        if (this.config.widget_plugin_src) {
+            tinymce.PluginManager.load('openmagewidget', this.config.widget_plugin_src);
+            this.openmagePluginsOptions.set('openmagewidget', {
+                'widget_window_url': this.config.widget_window_url
+            });
+        }
+
+        if (this.config.plugins) {
+            (this.config.plugins).each(function (plugin) {
+                tinymce.PluginManager.load(plugin.name, plugin.src);
+                self.openmagePluginsOptions.set(plugin.name, plugin.options);
+            });
+        }
+
+        tinymce.init(this.getSettings(mode));
+    },
+
+
+    getSettings: function (mode) {
```

Comment:
Can we put each section into its own function, so its easier to overwrite if needed?

Reply:
We can do this later in a separate PR